### DISCUSSION TITLE
Function reuse

### DISCUSSION THREAD

**Discussion by JonKragh** [microsoft/Power-Fx#367] [Ideas]
Hi - 

There was a Power Apps roadmap item that allowed for PowerFx function reuse. Something like VBA modules.

Is code reuse on the roadmap still? If yes, is there a new delivery estimate for this capability?

Thanks!

**Comment by MikeStall** [comment]
It is definitely on the roadmap. 

For Power Apps, some of this comes via Components (https://powerapps.microsoft.com/pl-pl/blog/enhanced-component-properties/)
We're still determining how to do this at the Power Fx SDK level. Are there any particular scenarios you'd like to try here?

**Reply by JonKragh** [reply]
The use case we see in power apps often is encapsulating a save routine to dataverse. Imagine pulling properties off of the screen and saving them to dataverse in one routine.

Currently, components do not have access to dataverse connections as far as I know so we fall back to encapsulating code in a hidden button that we invoke.

Something at the SDK level like VBA modules would be great.
